### PR TITLE
Fix bar marker color array legend style

### PR DESCRIPTION
**Pull request by etpinard**
fixes https://github.com/plotly/plotly.js/issues/1212

... bringing legend support for bar maker color arrays on par with scatter.

### CODE REVIEW

File: test/image/mocks/bar_marker_array.json [hunk 1]
```diff
@@ -23,5 +23,8 @@
       },
       "type": "bar"
     }
-  ]
+  ],
+  "layout": {
+    "showlegend": true
```
Comment (etpinard) [inline test/image/mocks/bar_marker_array.json]: which looks like

image

before https://github.com/plotly/plotly.js/pull/1289/commits/5d0d93025221ed8590b68aee45572b19181ed976

File: src/components/legend/style.js [hunk 2]
```diff
@@ -172,13 +172,14 @@ function styleBars(d) {
     barpath.exit().remove();
     barpath.each(function(d) {
         var w = (d.mlw + 1 || markerLine.width + 1) - 1,
```
Comment (alexcjohnson) [inline src/components/legend/style.js]: does this need `d` -> `d0` too? care to include line width and color arrays into the same mock?

Comment (etpinard) [inline src/components/legend/style.js]: Good :eye: 

Done in 8e718bfcd1265a06a011d5b1a786301ef11b474b

Comment (alexcjohnson) [inline src/components/legend/style.js]: for completeness:
> and color arrays

ie `marker.line.color`

Comment (etpinard) [inline src/components/legend/style.js]: Done in https://github.com/plotly/plotly.js/pull/1289/commits/81d3db1815055e83c9fa3411042fe38caeb5651c

File: test/image/mocks/bar_marker_array.json [hunk 3]
```diff
@@ -26,6 +26,12 @@
             3,
             2,
             3
+          ],
+          "color": [
+            "red",
+            "green",
+            "blue",
+            "yellow"
```
Comment (alexcjohnson) [inline test/image/mocks/bar_marker_array.json]: A thing of beauty :)

**Comment by alexcjohnson** [comment]
💃

**Comment by r1z1a** [comment]
I have hit an issue, I can recolor a single bar for multiple traces, the problem arises when you recolor the first bar, it seems to make whole legend that one color.

**Comment by nitramwin** [comment]
@etpinard, Im afraid this is occurring again in 2020. Could you take a look please?

**Comment by alexcjohnson** [comment]
@nitramwin can you open a new issue with a codepen showing the problem you've encountered?

**Comment by nitramwin** [comment]
@alexcjohnson This is my first time making a codepen - I hope this works. https://codepen.io/nitramwin/pen/mdEoxZo

**Comment by archmoj** [comment]
> @alexcjohnson This is my first time making a codepen - I hope this works. https://codepen.io/nitramwin/pen/mdEoxZo

Thanks @nitramwin. Tracked in #5285.

**Pull request merged**